client/src/server.rs

```diff
@@ -174,74 +233,91 @@ impl Server for ServerAPI {
     async fn register_client(
         &self,
         phone_number: String,
-        password: &str,
+        password: String,
         registration_request: RegistrationRequest,
-        _session: Option<&VerifiedSession>,
-    ) -> Result<RegistrationResponse, Self::Error> {
+        session: Option<&VerifiedSession>,
+    ) -> Result<RegistrationResponse, SignalClientError> {
         let payload = json!(registration_request);
-        let auth_header = BasicAuthorizationHeader::new(phone_number, 1, password.to_string());
+        let auth_header = BasicAuthorizationHeader::new(phone_number, 1, password);
         let mut res = self
-            .client
+            .http_client
             .post(REGISTER_URI)
             .body(payload)
             .header("Authorization", auth_header.encode())
             .await
-            .map_err(|err| SignalClientError::KeyError(err.to_string()))?;
-        Ok(from_slice(
-            res.body_bytes()
-                .await
-                .map_err(|_| RegistrationError::BadResponse)?
-                .as_ref(),
-        )
-        .map_err(|_| RegistrationError::BadResponse)?)
+            .map_err(|_| RegistrationError::NoResponse)?;
+        if res.status().is_success() {
+            Ok(from_slice(
+                res.body_bytes()
+                    .await
+                    .map_err(|err| RegistrationError::BadResponse(format!("{err}")))
+                    .expect("There")
```

Comment:
Please

Reply:
Skal være ?

Reply:
`.expect("There")` is not saying much @Laursen79